Mathlib/FieldTheory/Minpoly/IsConjRoot.lean

```diff
@@ -0,0 +1,306 @@
+/-
+Copyright (c) 2024 Jiedong Jiang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jiedong Jiang
+-/
+import Mathlib.FieldTheory.Minpoly.Basic
+import Mathlib.FieldTheory.Adjoin
+import Mathlib.FieldTheory.Normal
+
+/-!
+# Conjugate roots
+
+This file specialize to the property of conjugate roots. Given two elements over some ring `K`,
+these two elements are conjuagte roots over some `K` if they have the same minimal polynomial over
+`K`.
+
+## Main definitions
+
+* `IsConjRoot`: `IsConjRoot K x y` means `y` is a conjugate root of `x` over `K`.
+
+## Main results
+
+* `isConjRoot_iff_exists_algEquiv`: Let `L / K` be a normal field extension. For any two elements
+`x` and `y` in `L`, `IsConjRoot K x y` is equivalent to the existence of an algebra equivalence
+`σ : L ≃ₐ[K] L` such that `y = σ x`.
+* `not_mem_iff_exists_ne_and_isConjRoot`: Let `L / K` be a field extension. If `x` is a separable
+element over `K` and the minimal polynomial of `x` splits in `L`, then `x` is not in the `K` iff
+there exists a different conjugate root of `x` in `L` over `K`.
+
+## TODO
+Prove `IsConjRoot.smul`, if `x` and `y` are conjugate roots, then so are `r • x` and `r • y`.
+
+## Tags
+conjugate root, minimal polynomial
+-/
+
+
+open Polynomial minpoly IntermediateField
+
+variable {R K L S A B: Type*} [CommRing R] [CommRing S] [Ring A] [Ring B] [Field K] [Field L]
+variable [Algebra R S] [Algebra R A] [Algebra R B]
+variable [Algebra K S] [Algebra K L] [Algebra K A] [Algebra L S]
+
+variable (R) in
+/--
+We say that `x'` is a conjugate root of `x` over `K` if the minimal polynomial of `x` is the
+same as the minimal polynomial of `x'`.
+-/
+def IsConjRoot (x x' : A) : Prop := (minpoly R x) = (minpoly R x')
```

Comment:
```suggestion
We say that `y` is a conjugate root of `x` over `K` if the minimal polynomial of `x` is the
same as the minimal polynomial of `y`.
-/
def IsConjRoot (x y : A) : Prop := (minpoly R x) = (minpoly R y)
```
To match the module docstring...

Please also update below.

Reply:
Thank you! Now all of the variables are `x` and `y` instead of `x` and `x'`.